### DISCUSSION TITLE
Unable to Initialize WebView2.

### DISCUSSION THREAD

**Discussion by DavidM29** [unoplatform/uno#18769] [Q&A] [answered]
Hello, I am trying to display a pdf using WebView2, however calling `EnsureCoreWebView2Async` throws `%1 is not a valid Win32 application`, the exception is thrown if the `Source` property of `WebView2` is not specified, is that expected? Is the` Source` property mandatory? Also, when I DO specify the `Source`, an exception is not thrown, but the `CoreWebView2` still does not get initialized.

p.s.
I am testing this on Windows, so I am not sure if this question is relevant to Uno specifically, if it's not, I will open the discussion in WinAppSDK github repository.

**Comment by jeromelaban** [comment]
Thanks for the question! This is not an error we're aware of. Could you provide a repro of your issue?

**Reply by DavidM29** [reply]
Here is the repro:
[UnoApp1.zip](https://github.com/user-attachments/files/17796314/UnoApp1.zip)

**Reply by DavidM29** [reply]
I also tried just adding `WebView2` in xaml and setting the `Source` to `https://www.google.com` and nothing got displayed.

**Reply by DavidM29** [reply]
@jeromelaban I tried reproducing this in a WinUI only project and everything works as expected, so this is probably an uno issue.

**Reply by DavidM29** [reply]
Here is the WinUI only repro, I do not set the `source` property in the xaml and I call `EnsureCoreWebView2Async`, after that I navigate to a pdf file. everything works.
In uno, the same example throws an exception.
[App1.zip](https://github.com/user-attachments/files/17796970/App1.zip)

**Comment by MartinZikmund** [comment, marked as answer]
@DavidM29 The issue was previously reported at https://github.com/unoplatform/uno/issues/18552, you can check the thread for a workaround https://github.com/unoplatform/uno/issues/18552#issuecomment-2438052084 . Closing this discussion in favor of the linked issue 👍
